pkg/build/controller/strategy/util.go

```diff
@@ -536,6 +518,50 @@ func addVolumeMountToContainers(conts []corev1.Container, mount corev1.VolumeMou
 	return containers
 }
 
+// securityContextForBuild returns a security context that limits the requested
+// privileges because the build spec wanted an unprivileged build, or that sets
+// the pod to be privileged otherwise.
+func securityContextForBuild(vars []corev1.EnvVar) *corev1.SecurityContext {
+	privileged := true
+	for _, env := range vars {
+		if env.Name == "BUILD_PRIVILEGED" {
+			if b, err := strconv.ParseBool(env.Value); err == nil {
+				privileged = b
+			} else if i, err := strconv.Atoi(env.Value); err == nil {
+				privileged = i != 0
+			}
+		}
+	}
+	securityContext := &corev1.SecurityContext{
+		Privileged: &privileged,
+	}
+	if !privileged {
+		securityContext.Capabilities = &corev1.Capabilities{
+			Drop: []corev1.Capability{
+				"CAP_KILL",
+				"CAP_MKNOD",
+			},
+		}
```

Comment:
Do we need to run as uid/gid 0?
```suggestion
		securityContext.Capabilities = &corev1.Capabilities{
			Drop: []corev1.Capability{
				"CAP_KILL",
				"CAP_MKNOD",
			},
		}
		securityContext.RunAsUser = 0
		securityContext.RunAsGroup = 0
```

Reply:
That's the default for the struct, but you're right that there's no harm in making it explicit in the code.  Since that's what we're requesting whether or not we're privileged, should I just set these fields outside of the `if !privileged` block?

Reply:
Oh, those are pointers.  Got it.